### PR TITLE
Implement split_with_sizes backward for NT

### PR DESCRIPTION
**Pull request by jbschlosser**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #110647
* #110646

Needed internally. Note that `split_with_sizes()` for NT is currently supported only on `dim=-1`.

### CODE REVIEW

**Review by cpuhrsch — APPROVED** [review]
Stamp

File: torch/csrc/autograd/FunctionsManual.cpp [hunk 1]
```diff
@@ -2116,8 +2116,35 @@ Tensor _nested_split_with_sizes_backward(
     const std::vector<torch::autograd::Variable>& grads,
     c10::SymIntArrayRef split_sizes,
     int64_t dim,
-    const Tensor& self) {
-  return not_implemented("aten::split_with_sizes");
+    const Tensor& nt_sizes,
+    const at::TensorOptions& options) {
+  // add 1 to account for batch dim
+  dim = at::maybe_wrap_dim(dim, static_cast<int64_t>(nt_sizes.size(1)) + 1);
```
Comment (soulitzer) [inline torch/csrc/autograd/FunctionsManual.cpp:2122]: nit: maybe we don't need the static casts?

**Review by soulitzer — APPROVED** [review]

**Comment by kit1980** [comment]
I've cancelled the merge as https://github.com/pytorch/pytorch/pull/110404 is likely to be reverted.

**Pull request closed without merging**